DiscordChatExporter.Core/Discord/DiscordClient.cs

```diff
@@ -44,11 +45,18 @@ public class DiscordClient
                     : _token
             );
 
-            return await Http.Client.SendAsync(
+            var response = await Http.Client.SendAsync(
                 request,
                 HttpCompletionOption.ResponseHeadersRead,
                 innerCancellationToken
             );
+
+            if (_throttle != 0)
+            {
+                await Task.Delay(new Random().Next(_throttle, _throttle * 2), innerCancellationToken);
+            }
```

Comment:
1. Creating a `new Random()` defeats the purpose of using random. It should be `Random.Shared`
2. Why use random at all?
3. This should wait the throttle duration only if it hasn't already passed yet. I.e., you should subtract the time passed since the last request, because the work happening in between could have already introduced enough delay naturally

Reply:
Thanks for feedback

> Creating a new Random() defeats the purpose of using random. It should be Random.Shared

Random generator should be seeded from the system clock, but reusing Random instance is better according to documentation.

> 2\. Why use random at all?

Randomizing requests is common and very simple way to bypass bot detection

> Why not TimeSpan?

> This should wait the throttle duration only if it hasn't already passed yet. I.e., you should subtract the time passed since the last request

Because I used random, I didn't feel the need to perfectly time it.

But if we make randomization configurable (at least in CLI version), we should use TimeSpan.
For example `--throttle 50-100` would randomize requests between 50-100ms, but `--throttle 50` would send requests exactly after 50ms.

> Also, we should add this as an option to GUI as well

I agree. We should add it below "Parallel limit" in GUI->Global settings

Reply:
> I wouldn't bother with that. If Discord wanted to know that you're using DCE (it already knows anyway), they wouldn't use request timing to detect that.

You are right. As seen with other (non discord) related open source scraping projects, they have teams to detect them and the only way to bypass it is to create custom private scraper. 

So we should drop randomization feature to make the code simpler.

> If your only goal with randomization is to avoid "bot detection" then I would drop it since it's not a viable masking method.

To avoid bot detection is not the (main) goal. The goal is to never hit api limit if you set high enough throttle delay.